contrib/objectstore/subscriber/fieldexcluder/fieldexcluder.go

```diff
@@ -0,0 +1,61 @@
+package fieldexcluder
+
+import (
+	"reflect"
+	"strings"
+)
+
+func getStructFields(v reflect.Value) map[string]reflect.Value {
+	res := make(map[string]reflect.Value)
+	for i := 0; i < v.NumField(); i++ {
+		res[v.Type().Field(i).Name] = v.Field(i)
+	}
+	return res
+}
+
+func matchExcludedFields(field string, excludedFields []string) []string {
+	res := make([]string, 0)
+	for _, f := range excludedFields {
+		if strings.HasPrefix(f, field) {
+			if f == field {
+				return nil
+			}
+			res = append(res, f[len(field)+1:])
+		}
+	}
+	return res
+}
+
+func generateExcludeFieldsFunction(v reflect.Value, excludedFields []string) func(interface{}) map[string]interface{} {
+	if len(excludedFields) == 0 || v.Kind().String() != "struct" {
+		return nil
+	}
+
+	fieldMap := getStructFields(v)
+	fieldFuncsMap := make(map[string]func(interface{}) map[string]interface{})
+	for field := range fieldMap {
+		excludedSubFields := matchExcludedFields(strings.ToLower(field), excludedFields)
+		if excludedSubFields != nil {
+			fieldFuncsMap[field] = generateExcludeFieldsFunction(fieldMap[field], excludedSubFields)
+		}
+	}
+
+	return func(val interface{}) map[string]interface{} {
+		v := reflect.ValueOf(val)
```

Comment:
A general remark: usually reflect things are not good for performances. I think it would be better to write dedicated static marshallers.

Reply:
@safchain do you mean that I should write an `if` clause for every possible field (including subfields) of `flow.Flow`?

Reply:
yes It may be better for performances but I was more thinking about bunch of predefined marshaller like

```
MarshalOnlyLinkLayer(flow)
```

```
MarshalWithoutMetric(flow)
```

etc.

In the configuration you will specify the marshaller to use, so that no `if` at all. You specify the marshaller to use at the subscriber creation.

Reply:
@orozery can we (maybe) tag the fields somehow and use the tags for the marsheler. Moving from hard coded names of marshelers to tags in the description of the structs?